terraform/environments/cooker/macie.tf

```diff
@@ -0,0 +1,32 @@
+
+# Create macie account
+# When logging in as adminisitrator this is not needed but may be for our customers
+# Account name is MacieAccount. Publishes data every hour (choice are FIFTEEN_MINUTES, ONE_HOUR or SIX_HOURS)
+
+resource "aws_macie2_account" "macieaccess" { 
+   finding_publishing_frequency = "ONE_HOUR"
+   status                       = "ENABLED"
+ }
+
+# Now create a job
+# Job type options are ONE-TIME and SCHEDULED. If scheduled is chosen you need to select a frequency using schedule_frequency of daily_schedule, weekly_schedule or monthly_schedule
+# The buckets should be replaced but a list of buckets to examine. This could be limited to those in the account you are connected to or others to which you have access. 
+# Below there is a limited list of those from example shown. 
+# There are many other job options available and these should be looked at in terrform [job creation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/macie2_classification_job)
+
+resource "aws_macie2_classification_job" "test" {
+  job_type = "ONE_TIME"
+  name     = "JOBNAME"
+  s3_job_definition {
+    bucket_definitions {
+      account_id = "236861075084"
+      buckets    = [
```

Comment:
Buckets should not be hard coded, but should be retrieved programmatically.
EG.
```
aws_s3_bucket.example-bucket.id
```

Reply:
This is based on instructions in the terraform example

Reply:
Sure, but that doesn't mean that we should hard code bucket names and consider the matter closed. If nothing else, it doesn't set a particularly good example.

Reply:
Queried this in slack

Reply:
These can be retrieve in the following fashion:
`module.s3-bucket-lb.bucket.id`
`module.s3-bucket.bucket.id`

Reply:
They can be retrieve programmatically like so:
`aws_s3_bucket.resource-name.id`
However, I don't see any buckets belonging to this account in the code in `main`. If they're created for the account somewhere else, you could use a data call to get the information for the buckets and then reference it here.

You can see the terraform docs for the s3 bucket data call here: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket

You'd do something like this:
```
data "aws_s3_bucket" "example" {
  bucket_name = "my-bucket"
}
...
resource "aws_macie2_classification_job" "test" {
  ...
  s3_job_definition {
    bucket_definitions {
      buckets = [data.aws_s3_bucket.example.id]
    }
  }
}
```

Reply:
This has now been amended and comments included above.
To do this I had to add new data fields to platform_data which is included in this release.

Reply:
This is better but not yet complete; you have data calls for buckets but I don't see the actual data calls in terraform.

EG.
```
data "aws_s3_bucket" "this-is-an-example" {
  bucket_name = "this-is-an-example"
}
```
Without data calls, there's nothing to retrieve for `data.aws_s3_bucket.bucket1.id`.

Reply:
The above didn't quite work but I got an updated version in place. data is in the data.tf terraform code.